packages/crayons-core/src/components/form/form.tsx

```diff
@@ -0,0 +1,422 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import { Component, Prop, State, Element, h, Method } from '@stencil/core';
+import {
+  FormValues,
+  FormTouched,
+  FormErrors,
+  FormUtils,
+  FormProps,
+  FormSubmit,
+} from './form-declaration';
+import {
+  validateYupSchema,
+  prepareDataForValidation,
+  yupToFormErrors,
+  generateDynamicInitialValues,
+  generateDynamicValidationSchema,
+} from './form-util';
+import { debounce, createGuid } from '../../utils';
+
+@Component({
+  tag: 'fw-form',
+  shadow: true,
+})
+export class Form {
+  @Element() el!: any;
+
+  private controls: any;
+
+  /**
+   * Initial field values of the form. It is an object with keys pointing to field name
+   */
+  @Prop() initialValues?: any = {};
+
+  /** Validate the form's values with an async function.
+   * Should return a Promise which resolves to an errors object.
+   * The keys in the errors object must match with the field names.
+   */
+  @Prop() validate?: any;
+
+  /**
+   * Schema to render Dynamic Form. Contains an array of fields pointing to each form control.
+   * Please see the usage reference for examples.
+   */
+  @Prop() formSchema?: any = {};
+
+  /**
+   * YUP based validation schema for handling validation
+   */
+  @Prop() validationSchema?: any = {};
+
+  /** Tells Form to validate the form on each input's onInput event */
+  @Prop() validateOnInput? = true;
+  /** Tells Form to validate the form on each input's onBlur event */
+  @Prop() validateOnBlur? = true;
+
+  /** The number of milliseconds to delay before doing validation on Input */
+  @Prop() wait = 400;
+
+  /**
+   * Id to uniquely identify the Form. If not set, a random Id will be generated.
+   */
+  @Prop() formId = createGuid();
+  @State() isValid = false;
+  @State() isValidating = false;
+  @State() isSubmitting = false;
+
+  @State() focused: keyof FormValues = null;
+  @State() values: FormValues = {} as any;
+  @State() touched: FormTouched<FormValues> = {} as any;
+  @State() errors: FormErrors<FormValues> = {} as any;
+
+  @State() formValidationSchema;
+  @State() formInitialValues;
+
+  private debouncedHandleInput: any;
+  private handleInputListener: any;
+  private handleFocusListener: any;
+  private handleBlurListener: any;
+  private handleChangeListener: any;
+
+  async componentWillLoad() {
+    this.debouncedHandleInput = debounce(this.handleInput, this, this.wait);
+
+    this.handleInputListener = this.el.addEventListener(
+      'fwFormInput',
+      this.debouncedHandleInput
+    );
+    this.handleBlurListener = this.el.addEventListener(
+      'fwFormBlur',
+      this.handleBlur
+    );
+    this.handleChangeListener = this.el.addEventListener(
+      'fwFormChange',
+      this.handleInput
```

Comment:
Handle change and input seem to be pointing to the same method... then why do we need two separate events to be captured?